Rakefile

```diff
@@ -25,16 +25,8 @@ namespace :test do
     task.pattern = 'test/**/*_test.rb'
   end
 
-  ExerciseTestTask.new :exercise do |task|
-    task.description = "Run the tests for a specific exercise"
-  end
-
   desc 'Run the tests for all exercises'
   task :exercises do
-    Dir.foreach('exercises').each do |exercise|
-      next if %w(. ..).include?(exercise)
-
-      Rake::Task['test:exercise'].execute(exercise: exercise)
-    end
+    ExerciseTests.run_all
```

Comment:
This seems like the wrong level of abstraction.
You're hiding the similarity with line 14.